src/main/java/org/cyclonedx/gradle/utils/EnvironmentUtils.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * This file is part of CycloneDX Gradle Plugin.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) OWASP Foundation. All Rights Reserved.
+ */
+package org.cyclonedx.gradle.utils;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class EnvironmentUtils {
+
+    /**
+     * Checks if a string is null or blank.
+     * @param value the string to check
+     * @return true if the string is null or blank, false otherwise
+     */
+    private static boolean isNullOrBlank(String value) {
+        return value == null || value.trim().isEmpty();
+    }
+
+    /**
+     * Get the URI of the current build from the environment variables set on common build systems like GitHub Actions, GitLab CI, etc.
+     * @return the URI of the current build or null if it cannot be determined
+     */
+    public static String getBuildURI() {
+        // GitHub Actions
```

Comment:
if you extract each code block into a method, you could name the method in a similar fashion as your comment, and drop the comment. for example

```
        return Optional.ofNullable(fromGithubActions())
                       .orElseGet(() -> Optional.ofNullable(fromGitlabCI()))
                       .orElseGet(() -> Optional.ofNullable(fromJenkins()))
                         ...
                       .orElse(null); 
```

```
private static final String fromGithubActions() {
        String githubServerUrl = System.getenv("GITHUB_SERVER_URL");
        String githubRepository = System.getenv("GITHUB_REPOSITORY");
        String githubRunId = System.getenv("GITHUB_RUN_ID");

        if (!isNullOrBlank(githubServerUrl) && !isNullOrBlank(githubRepository) && !isNullOrBlank(githubRunId)) {
            return String.format("%s/%s/actions/runs/%s", githubServerUrl, githubRepository, githubRunId);
        }
        
        return null;
}

```

Reply:
Thanks for the feedback! The code has been updated and is a little more elegant now ;)